spyder/plugins/ipythonconsole/widgets/main_widget.py

```diff
@@ -0,0 +1,479 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright © Spyder Project Contributors
+# Licensed under the terms of the MIT License
+# (see spyder/__init__.py for details)
+
+"""
+IPython Console main widget based on QtConsole.
+"""
+
+# Standard library imports
+import os
+import os.path as osp
+import sys
+
+# Third-party imports
+from jupyter_core.paths import jupyter_config_dir, jupyter_runtime_dir
+from qtpy.QtCore import Slot
+from qtpy.QtGui import QColor
+from qtpy.QtWebEngineWidgets import WEBENGINE
+from qtpy.QtWidgets import (QHBoxLayout, QLabel, QVBoxLayout, QWidget)
+
+
+# Local imports
+from spyder.api.translations import get_translation
+from spyder.api.widgets.main_widget import PluginMainWidget
+from spyder.config.base import (
+    get_conf_path, get_home_dir, running_under_pytest)
+from spyder.plugins.ipythonconsole.widgets.client import ClientWidgetActions
+from spyder.plugins.ipythonconsole.widgets import (
+    ClientWidget, ConsoleRestartDialog, KernelConnectionDialog,
+    PageControlWidget)
+from spyder.py3compat import is_string, to_text_string, PY2, PY38_OR_MORE
+from spyder.utils import programs, sourcecode
+from spyder.utils.palette import QStylePalette
+from spyder.widgets.browser import FrameWebView
+from spyder.widgets.findreplace import FindReplace
+from spyder.widgets.tabs import Tabs
+
+
+# Localization
+_ = get_translation('spyder')
+
+# =============================================================================
+# ---- Constants
+# =============================================================================
+MAIN_BG_COLOR = QStylePalette.COLOR_BACKGROUND_1
+
+class IPythonConsoleWidgetActions:
+    # Clients creation
+    CreateNewClient = 'create_new_client_action'
+    CreateCythonClient = 'create_cython_client_action'
+    CreateSymPyClient = 'create_sympy_client_action'
+    CreatePyLabClient = 'create_pylab_client_action'
+    
+    # Current console actions
+    ClearConsole = 'clear_console_action'
+    ClearLine = 'clear_line'
+    ConnectToKernel = 'connect_to_kernel_action'
+    Interrupt = 'interrupt_action'
+    InspectObject = 'inspect_object_action'
+    Restart = 'restart_action'
+    RemoveAllVariables = 'remove_all_variables_action'
+    ResetNamespace = 'reset_namespace_action'
+
+    # Tabs
+    RenameTab = 'rename_tab_action'
+    NewTab = 'new_tab_action'
+
+    # Variables display
+    ArrayInline = 'arrya_iniline_action'
+    ArrayTable = 'arrya_table_action'
```

Comment:
```suggestion
    ArrayInline = 'array_iniline_action'
    ArrayTable = 'array_table_action'
```

Minor correction because I spotted it on the fly.